### PR TITLE
My VA - date formatting helpers cleanup + tests

### PR DESCRIPTION
**Pull request by allisonlu**
## Summary
This PR 
- updates remaining remnants of `moment` instances to regular Date objects or `date-fns`
- created a `date-formatting` folder for their unit tests
- added unit tests for helpers and timezone.js 

No UI changes. This is tech debt cleanup.

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/66858
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/21395

## Testing done
- all existing unit and e2e still pass
- locally, visually

## Screenshots
Screenshot 2023-12-21 at 1 40 23 PM

## What areas of the site does it impact?
My VA - Health care - Next appt card

## Acceptance criteria
- [ ] Next appointment card still appears correctly

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Comment by adamwhitlock1** [comment]
Probably a good idea to update moment-tz lib code as well.

The newest version is here:
https://momentjs.com/downloads/moment-timezone-with-data.js

It looks like vets-api is also using the 2023c version of the timezone data now at least from looking at this PR: 
https://github.com/department-of-veterans-affairs/vets-api/pull/14328 (unfortunately its not actually letting me see the file diff from the PR though...)

Non-blocking, but just thought it warranted bringing up.

File: src/applications/personalization/appointments/actions/utils.js [hunk 1]
```diff
@@ -29,12 +25,7 @@ const getStagingID = facilityID => {
 const transformAppointment = appointment => {
   const isVideo = appointment.attributes?.kind === 'telehealth';
   const facilityStagingId = getStagingID(appointment.attributes.locationId);
-  const timezone = getTimezoneBySystemId(facilityStagingId)?.timezone;
-  const date = appointment.attributes.start;
-  const startsAt = (timezone
-    ? moment(date).tz(timezone)
-    : moment(date)
-  ).format();
+  const startsAt = appointment.attributes.start;
```
Comment (adamwhitlock1) [inline src/applications/personalization/appointments/actions/utils.js:28]: Originally the `startsAt` took into account the local system based timezone and adjusted the time based on that, but now we are just using the raw `appointment.attributes.start`

Is this done for a specific reason? I'm not quite following the change here, as it seems like it wouldn't adjust appropriately anymore.

Comment (allisonlu) [reply src/applications/personalization/appointments/actions/utils.js:28]: I wanted to remove any dependency on the `moment` library, and was trying to look into an equivalent to `moment-tz` but turns out that the [Date object operates on the user's host environment/device](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#date_components_and_time_zones). 

We're using the `startsAt` variable in AppointmentsCard component, where the [raw string is converted to a Date object.](https://github.com/department-of-veterans-affairs/vets-website/blob/My-VA/allison/66858/timezone-cleanup/src/applications/personalization/dashboard/components/health-care/AppointmentsCard.jsx#L10).

Let me know if you have any other ideas!

**Comment by allisonlu** [comment]
> It looks like vets-api is also using the 2023c version of the timezone data now at least from looking at this PR:
> [department-of-veterans-affairs/vets-api#14328](https://github.com/department-of-veterans-affairs/vets-api/pull/14328) (unfortunately its not actually letting me see the file diff from the PR though...)

This is extremely interesting 🧐 I'm not able to see the diff or changed files either. 

I'm thinking of removing the moment-tz file entirely, but only after [cleaning up old appointment v0 code in this ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/72297)

File: src/applications/personalization/appointments/actions/utils.js (hunk 1 above)
Comment (adamwhitlock1) [reply src/applications/personalization/appointments/actions/utils.js:28]: Ok that makes sense and helps me understand what is happening here. Thanks!

:shipit:

**Review by adamwhitlock1 — APPROVED** [review]

**Pull request merged**